File: librarymanager/librarymanager-be/src/main/java/ru/elenacherev/librarymanager/api/BookUsingApi.java
package ru.elenacherev.librarymanager.api;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import ru.elenacherev.librarymanager.api.dto.BookUsing;

public interface BookUsingApi {

	BookUsing getBookUsing(
			@NotNull Long bookusingid
	);

	Page<BookUsing> searchBookUsings(
			Pageable pageable
	);

	BookUsing createBookUsing(
			@Valid BookUsing bookUsing
	);

	BookUsing saveBookUsing(
			@NotNull Long bookUsingId, 
			@Valid BookUsing bookUsing
	);

}

File: librarymanager-api/src/main/java/ru/elenacherev/librarymanager/api/BookUsingApi.kt
package ru.elenacherev.librarymanager.api

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import ru.elenacherev.librarymanager.api.dto.BookUsing
import java.util.*
import javax.validation.Valid

interface BookUsingApi {
    fun getBookUsing(
        bookusingid: UUID
    ): BookUsing?

    fun searchBookUsings(
        pageable: Pageable
    ): Page<BookUsing>

    fun createBookUsing(
        bookUsing: @Valid BookUsing
    ): BookUsing

    fun saveBookUsing(
        bookUsingId: UUID,
        bookUsing: @Valid BookUsing
    ): BookUsing
}